rules/scripts/bitstreams_workspace_test.py

```diff
@@ -107,5 +107,116 @@ def test_write_build_file(self):
         os.walk.assert_called_once_with('cache/abcd')
 
 
+class TestFetchAvailableBitstreams(unittest.TestCase):
+    """
+    The BitstreamCache.GetBitstreamsAvailable method calls the XML API on
+    the root level of the GCP bucket to get the list of files in the
+    bitstream cache. If this list is sufficiently long, the response will
+    be paginated, as indicated by the <IsTruncated> tag. For p pages,
+    GetBitstreamsAvailable is expected to make p+1 calls to the API (using
+    BitstreamCache.Get). The last call is to get latest.txt which indicates
+    the latest bitstream.
+
+    The XML response elements are documented here:
+    https://cloud.google.com/storage/docs/xml-api/get-bucket-list#response_body_elements
+    """
+
+    def setUp(self):
+        self.cache = BitstreamCache('/', '/tmp/cache/opentitan-bitstreams',
```

Comment:
I'm not sure about setting up a single cache instance that's shared between all the tests since it makes the tests kind of interdependent. In this particular case, it works, but I don't know if this is best practice.

Reply:
I believe `setUp` is called before each test method, so this will do the right thing :)